packages/eslint-plugin-xcritical/src/rules/typescript.js

```diff
@@ -15,6 +15,29 @@ const typescriptRules = {
   '@typescript-eslint/no-type-alias': 'off',
   '@typescript-eslint/no-explicit-any': 'warn',
   '@typescript-eslint/quotes': ['error', 'single'],
-  '@typescript-eslint/explicit-function-return-type': ['error', { allowExpressions: true, allowTypedFunctionExpressions: true, allowHigherOrderFunctions: true }],
+  '@typescript-eslint/no-use-before-define': ['error', { functions: false }],
+  '@typescript-eslint/no-unnecessary-condition': ['error', { ignoreRhs: true }],
+  '@typescript-eslint/no-inferrable-types': 0, // this is a temporary solution due to problems in TS (see: https://github.com/microsoft/TypeScript/pull/30593)
+  '@typescript-eslint/restrict-template-expressions': 0,
+  '@typescript-eslint/space-before-function-paren': 0,
+  '@typescript-eslint/explicit-function-return-type': [1, {
+    allowExpressions: true,
+    allowTypedFunctionExpressions: true,
+    allowHigherOrderFunctions: true,
+  }],
+  'import/extensions': ['error', 'never', { svg: 'always' }],
+  'import/no-extraneous-dependencies': 0,
+  '@typescript-eslint/require-await': 0,
+  '@typescript-eslint/promise-function-async': ['warn'],
+  '@typescript-eslint/no-non-null-assertion': ['warn'],
+  '@typescript-eslint/prefer-readonly-parameter-types': 0,
+  '@typescript-eslint/no-unsafe-assignment': 0,
+  '@typescript-eslint/no-unsafe-call': 0,
+  '@typescript-eslint/no-unsafe-return': 0,
+  '@typescript-eslint/no-unsafe-member-access': 0,
+  '@typescript-eslint/explicit-module-boundary-types': ['warn'],
+  '@typescript-eslint/init-declarations': 0,
+  '@typescript-eslint/default-param-last': ['warn'],
+  '@typescript-eslint/no-non-null-asserted-optional-chain': ['warn'],
```

Comment:
How about adding this: `'@typescript-eslint/no-dynamic-delete': 0`?

Reply:
What you mean?